staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -50,7 +60,9 @@ func (o *Options) Validate() []error {
 
 // AddFlags add logging-format flag
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, "Set log format")
+	unsupportedFlags := fmt.Sprintf("--%s", strings.Join(unsupportedLoggingFlags(), ", --"))
+	formats := fmt.Sprintf(`"%s"`, strings.Join(logRegistry.List(), `", "`))
+	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Set format of logs written by component. Options: %s. Non default formats don't support klog flags like: %s.", formats, unsupportedFlags))
```

Comment:
```suggestion
	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, unsupportedFlags))
```

Since you very cleverly got a list of flags, consider actually checking at run time to see if the user is trying to use one of those flags while setting a non-default output? It's a long list, I think it'll be very easy to accidentally use one.

Reply:
With this PR I want to unblock PRs that use this flag. As discussed earlier with @mtaufen, we definitely want to implement validating conflicting flags. For now we decided to document it allowing effort to move forward

Reply:
Sure, the validation part doesn't need to block this PR. Adding the "alpha" notice is important though.